container/navigation.go

```diff
@@ -0,0 +1,207 @@
+package container
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/layout"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Navigation container is used to provide your application with a control bar and an area for content objects.
+// Objects can be any CanvasObject, and only the most recent one will be visible.
+//
+// Since: 2.7
+type Navigation struct {
+	widget.BaseWidget
+	level  int
+	stack  fyne.Container
+	titles []string
+
+	Root   fyne.CanvasObject
+	Title  string
+	OnBack func()
+	OnNext func()
+}
+
+// NewNavigation creates a new navigation container with a given root object.
+//
+// Since: 2.7
+func NewNavigation(root fyne.CanvasObject) *Navigation {
+	return NewNavigationWithTitle(root, "")
+}
+
+// NewNavigationWithTitle creates a new navigation container with a given root object and a default title.
+//
+// Since: 2.7
+func NewNavigationWithTitle(root fyne.CanvasObject, s string) *Navigation {
+	var nav *Navigation
+	nav = &Navigation{
+		Root:   root,
+		Title:  s,
+		OnBack: func() { _ = nav.Back() },
+		OnNext: func() { _ = nav.Forward() },
+	}
+	return nav
+}
+
+// Push puts the given object on top of the navigation stack and hides the object below.
+//
+// Since: 2.7
+func (nav *Navigation) Push(obj fyne.CanvasObject) {
+	s := nav.Title
+	if nav.level > 0 {
+		s = nav.titles[nav.level-1]
+	}
+	nav.PushWithTitle(obj, s)
+}
+
+// PushWithTitle puts the given CanvasObject on top, hides the object below, and uses the given title as label text.
+//
+// Since: 2.7
+func (nav *Navigation) PushWithTitle(obj fyne.CanvasObject, s string) {
+	obj.Show()
+	objs := nav.stack.Objects[:nav.level]
+	if len(objs) > 0 {
+		objs[len(objs)-1].Hide()
+	}
+	nav.stack.Objects = append(objs, obj)
+	nav.titles = append(nav.titles[:nav.level], s)
+	nav.level++
+	nav.Refresh()
+}
+
+// Back returns the top level CanvasObject, adjusts the title accordingly, and disabled the back button
+// when no more objects are left to go back to.
+//
+// Since: 2.7
+func (nav *Navigation) Back() fyne.CanvasObject {
+	if nav.level == 0 || nav.level == 1 && nav.Root != nil {
+		return nil
+	}
+
+	objs := nav.stack.Objects
+	objs[nav.level-1].Hide()
+	if nav.level > 1 {
+		objs[nav.level-2].Show()
+	}
+
+	nav.level--
+	nav.Refresh()
+
+	return objs[nav.level]
+}
+
+// Forward shows the next object in the stack again.
+//
+// Since: 2.7
+func (nav *Navigation) Forward() fyne.CanvasObject {
+	if nav.level >= len(nav.stack.Objects) {
+		return nil
+	}
+
+	nav.stack.Objects[nav.level-1].Hide()
+	nav.stack.Objects[nav.level].Show()
+	nav.level++
+
+	return nav.stack.Objects[nav.level-1]
+}
+
+// SetTitle changes the navigation title and the title for the current object.
+//
+// Since: 2.7
+func (nav *Navigation) SetTitle(s string) {
```

Comment:
This function makes me a little nervous - what it does varies with what the current level is but the field `Title` does not vary... 

I think for consistency this should be setting the root title. Perhaps another method that more clearly indicates it is setting the title of a level.